### ISSUE TITLE
[NotSupportedException: String access is not supported] when await UnityWebRequest with DownloadHandlerAssetBundle 

### ISSUE THREAD

**Issue by ReOcto** [Cysharp/UniTask#110]
Example code:
`     await UnityWebRequestAssetBundle.GetAssetBundle("invalidUrl").SendWebRequest();`

Exception occurs when UnityWebRequest with DownloadHandlerAssetBundle fails. 
UniTask tries to create UnityWebRequestException and apparently it tries to access downloadHandler.text:

```
NotSupportedException: String access is not supported
UnityEngine.Networking.DownloadHandlerFile.GetText () (at /Users/builduser/buildslave/unity/build/Modules/UnityWebRequest/Public/DownloadHandler/DownloadHandler.bindings.cs:252)
UnityEngine.Networking.DownloadHandler.get_text () (at /Users/builduser/buildslave/unity/build/Modules/UnityWebRequest/Public/DownloadHandler/DownloadHandler.bindings.cs:54)
Cysharp.Threading.Tasks.UnityWebRequestException..ctor (UnityEngine.Networking.UnityWebRequest unityWebRequest) (at Library/PackageCache/com.cysharp.unitask@73a5ff6648/Runtime/UnityWebRequestException.cs:34)
Cysharp.Threading.Tasks.UnityAsyncExtensions+UnityWebRequestAsyncOperationConfiguredSource.MoveNext () (at Library/PackageCache/com.cysharp.unitask@73a5ff6648/Runtime/UnityAsyncExtensions.cs:1475)
Cysharp.Threading.Tasks.Internal.PlayerLoopRunner.RunCore () (at Library/PackageCache/com.cysharp.unitask@73a5ff6648/Runtime/Internal/PlayerLoopRunner.cs:154)
UnityEngine.Debug:LogException(Exception)
Cysharp.Threading.Tasks.Internal.<>c:<.ctor>b__9_0(Exception) (at Library/PackageCache/com.cysharp.unitask@73a5ff6648/Runtime/Internal/PlayerLoopRunner.cs:25)
Cysharp.Threading.Tasks.Internal.PlayerLoopRunner:RunCore() (at Library/PackageCache/com.cysharp.unitask@73a5ff6648/Runtime/Internal/PlayerLoopRunner.cs:168)
Cysharp.Threading.Tasks.Internal.PlayerLoopRunner:Update() (at Library/PackageCache/com.cysharp.unitask@73a5ff6648/Runtime/Internal/PlayerLoopRunner.cs:127)
Cysharp.Threading.Tasks.Internal.PlayerLoopRunner:Run() (at Library/PackageCache/com.cysharp.unitask@73a5ff6648/Runtime/Internal/PlayerLoopRunner.cs:94)
```

**Comment by Strepto** [comment]
Could consider an implementation looking like this for getting response errors:

```cs
        private static string getResponseText(UnityWebRequest response)
        {
            if (response.downloadHandler == null)
                return null;

            if (response.downloadHandler is DownloadHandlerBuffer)
                return response.downloadHandler.text;

            return $"ResponseText not available for {response.downloadHandler.GetType().Name}";
        }
```

**Comment by neuecc** [comment]
Thanks for reporting and thanks @Strepto idea.
I've added check of `DownloadHandlerBuffer` in `UnityWebRequestException`.

**Comment by neuecc** [comment]
fixed at 2.0.25.
